src/Schema/Column/BigIntColumnSchema.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Db\Schema\Column;
+
+use Yiisoft\Db\Expression\ExpressionInterface;
+use Yiisoft\Db\Schema\SchemaInterface;
+
+use function is_int;
+
+use const PHP_INT_MAX;
+use const PHP_INT_MIN;
+
+class BigIntColumnSchema extends AbstractColumnSchema
+{
+    public function __construct(
+        string $type = SchemaInterface::TYPE_BIGINT,
+        string|null $phpType = SchemaInterface::PHP_TYPE_STRING,
+    ) {
+        parent::__construct($type, $phpType);
+    }
+
+    public function dbTypecast(mixed $value): int|string|ExpressionInterface|null
+    {
+        if (is_int($value)) {
+            return $value;
+        }
+
+        return match ($value) {
+            null, '' => null,
+            false => 0,
+            default => $value instanceof ExpressionInterface
```

Comment:
This will not be optimal. One of the optimization ideas is to move this check to the last plan. First we check standard situations, then the rest. In the case with `bigint` we check if the value is `string`, `null` or `int` and if not we check other cases. Refactored.